apps/drec-ui/src/apps/device-group/view/containers/modals/UnreserveSelected/UnreserveSelected.effects.ts

```diff
@@ -0,0 +1,43 @@
+import { GenericModalProps } from '@energyweb/origin-ui-core';
+import { useUnreserveSelectedGroups } from '../../../../data';
+import { useUnreserveSelectedModalLogic } from '../../../../logic';
+import {
+    useDeviceGroupModalsStore,
+    useDeviceGroupModalsDispatch,
+    DeviceGroupModalsActionsEnum
+} from '../../../context';
+
+export const useUnreserveSelectedEffects = () => {
+    const {
+        unreserveSelected: { open, selected }
+    } = useDeviceGroupModalsStore();
+    const dispatchModals = useDeviceGroupModalsDispatch();
+
+    const closeModal = () => {
+        dispatchModals({
+            type: DeviceGroupModalsActionsEnum.UNRESERVE,
+            payload: {
+                open: false,
+                selected: []
+            }
+        });
+    };
+
+    const unreserveSelectedHandler = useUnreserveSelectedGroups(selected, closeModal);
+
+    const unreserveHandler = () => {
+        dispatchModals({
+            type: DeviceGroupModalsActionsEnum.UNRESERVE,
+            payload: { open: false, selected: [] }
+        });
+        unreserveSelectedHandler();
```

Comment:
you have to first call `unreserveSelectedHandler` and then dispatchModals. Because even thought dispatchModals is async - it can cause an issue with unstable state of `selected`. So make sure you first call the handler and only then close modal and reset state.
Also you can replace whole `dispatchModals...` by `closeModal()`